### ISSUE TITLE
[Bug 🐞]: Terraform Plugin Crash on Deployment 

### ISSUE THREAD

**Issue by Parkers145** [threefoldtech/terraform-provider-grid#513]
### What happened?

When Deploying using terraform he plugin crashed and generated a error that said it should be reported error in log output 

### What did you expect?

Successful deployment of this main.tf 

```

variable "MNEMONICS" {
  type        = string
  description = "The mnemonic phrase used to generate the seed for the node."
}

variable "NETWORK" {
  type        = string
  default     = "main"
  description = "The network to connect the node to."
}

variable "SSH_KEY" {
  type = string
}


terraform {
  required_providers {
    grid = {
      source = "threefoldtech/grid"
    }
  }
}

provider "grid" {
    mnemonics = "${var.MNEMONICS}"
    network   = "${var.NETWORK}"  
}
resource "grid_network" "net1" {
    nodes = [8]
    ip_range = "10.0.0.0/16"
    name = "Net1"
    description = "MyNetwork1"
    add_wg_access = "true"
}
resource "grid_deployment" "D1" { 
  node = 8 
  network_name = grid_network.net1.name 
  disks { 
	 name = "Disk1" 
	 size = "25" 
  } 
  disks { 
	 name = "Disk2" 
	 size = "30" 
  } 
    vms { 
    name = "VM1" 
    description = "MyVm1" 
    flist = "https://hub.grid.tf/tf-official-vms/ubuntu-20.04-lts.flist" 
    cpu = "4" 
    publicip = "true" 
    publicip6 = "true" 
    memory = "4096" 
    mounts { 
	 disk_name = "Disk1" 
	 mount_point = "/data1" 
	}
	mounts { 
	 disk_name = "Disk2" 
	 mount_point = "/data2" 
	}
	planetary = "true" 
    env_vars = { 
      SSH_KEY = "${var.SSH_KEY}" 
    } 
  } 
} 

output "wg_config1" { 
  value = grid_network.net1.access_wg_config 
} 
output "public_ip1" {
 value = grid_deployment.D1.vms[0].computedip 
} 
output "public_ip61" { 
  value = grid_deployment.D1.vms[0].computedip6 
} 
output "ygg_ip1" { 
   value = grid_deployment.D1.vms[0].ygg_ip 
} 
```

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
root@SmithTerraformr:/deployments/desktop# terraform apply  -var-file="/deployments/main.tfvars"

Terraform used the selected providers to generate the following execution plan.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # grid_deployment.D1 will be created
  + resource "grid_deployment" "D1" {
      + id                = (known after apply)
      + ip_range          = (known after apply)
      + name              = "vm"
      + network_name      = "Net1"
      + node              = 8
      + solution_provider = 0
      + solution_type     = "Virtual Machine"

      + disks {
          + name = "Disk1"
          + size = 25
        }
      + disks {
          + name = "Disk2"
          + size = 30
        }

      + vms {
          + computedip  = (known after apply)
          + computedip6 = (known after apply)
          + corex       = false
          + cpu         = 4
          + description = "MyVm1"
          + env_vars    = {
              + "SSH_KEY" = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIFqf0+CsLH41KOlxesTlTkZRghWm4nEkHL9448XDa1HW parker@smithtac"
            }
          + flist       = "https://hub.grid.tf/tf-official-vms/ubuntu-20.04-lts.flist"
          + ip          = (known after apply)
          + memory      = 4096
          + name        = "VM1"
          + planetary   = true
          + publicip    = true
          + publicip6   = true
          + ygg_ip      = (known after apply)

          + mounts {
              + disk_name   = "Disk1"
              + mount_point = "/data1"
            }
          + mounts {
              + disk_name   = "Disk2"
              + mount_point = "/data2"
            }
        }
    }

  # grid_network.net1 will be created
  + resource "grid_network" "net1" {
      + access_wg_config   = (known after apply)
      + add_wg_access      = true
      + description        = "MyNetwork1"
      + external_ip        = (known after apply)
      + external_sk        = (known after apply)
      + id                 = (known after apply)
      + ip_range           = "10.0.0.0/16"
      + name               = "Net1"
      + node_deployment_id = (known after apply)
      + nodes              = [
          + 8,
        ]
      + nodes_ip_range     = (known after apply)
      + public_node_id     = (known after apply)
      + solution_type      = "Network"
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + public_ip1  = (known after apply)
  + public_ip61 = (known after apply)
  + wg_config1  = (known after apply)
  + ygg_ip1     = (known after apply)

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

grid_network.net1: Creating...
╷
│ Error: Request cancelled
│
│   with grid_network.net1,
│   on main.tf line 30, in resource "grid_network" "net1":
│   30: resource "grid_network" "net1" {
│
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-grid_v1.6.3 plugin:

fatal error: runtime: out of memory

runtime stack:
runtime.throw({0xe3c1a2?, 0x1338800000?})
        runtime/panic.go:992 +0x71
runtime.sysMap(0xc000800000, 0xc000133e90?, 0xc000133ef8?)
        runtime/mem_linux.go:189 +0x11b
runtime.(*mheap).grow(0x1529640, 0x99c215?)
        runtime/mheap.go:1413 +0x225
runtime.(*mheap).allocSpan(0x1529640, 0x99c215, 0x0, 0x1)
        runtime/mheap.go:1178 +0x171
runtime.(*mheap).alloc.func1()
        runtime/mheap.go:920 +0x65
runtime.systemstack()
        runtime/asm_amd64.s:469 +0x49

goroutine 66 [running]:
runtime.systemstack_switch()
        runtime/asm_amd64.s:436 fp=0xc0004a5008 sp=0xc0004a5000 pc=0x461e80
runtime.(*mheap).alloc(0x1000000ac3afe?, 0xc0004a50d0?, 0xc5?)
        runtime/mheap.go:914 +0x65 fp=0xc0004a5050 sp=0xc0004a5008 pc=0x4268c5
runtime.(*mcache).allocLarge(0x44acb2?, 0x1338428100, 0x1)
        runtime/mcache.go:213 +0x85 fp=0xc0004a50a0 sp=0xc0004a5050 pc=0x416da5
runtime.mallocgc(0x1338428100, 0xd1e0a0, 0x1)
        runtime/malloc.go:1096 +0x5a5 fp=0xc0004a5118 sp=0xc0004a50a0 pc=0x40d1a5
runtime.newarray(0xd78360?, 0xc0003d0f01?)
        runtime/malloc.go:1291 +0x52 fp=0xc0004a5140 sp=0xc0004a5118 pc=0x40d6b2
reflect.unsafe_NewArray(0xc000586180?, 0xc00027e5f4?)
        runtime/malloc.go:1296 +0x19 fp=0xc0004a5160 sp=0xc0004a5140 pc=0x45db19
reflect.MakeSlice({0xf92b68, 0xc8f880}, 0x99c21408, 0x99c21408)
        reflect/value.go:2911 +0xac fp=0xc0004a5190 sp=0xc0004a5160 pc=0x497e2c
github.com/centrifuge/go-substrate-rpc-client/v4/scale.Decoder.DecodeIntoReflectValue({{0xf85880?, 0xc0001b9170?}}, {0xc8f880?, 0xc0001600c8?, 0xc0004a5438?})
        github.com/centrifuge/go-substrate-rpc-client/v4@v4.0.5/scale/codec.go:436 +0x48a fp=0xc0004a53a0 sp=0xc0004a5190 pc=0xac448a
github.com/centrifuge/go-substrate-rpc-client/v4/scale.Decoder.Decode({{0xf85880?, 0xc0001b9170?}}, {0xc000586180?, 0xc0001600c8})
        github.com/centrifuge/go-substrate-rpc-client/v4@v4.0.5/scale/codec.go:325 +0x28d fp=0xc0004a5448 sp=0xc0004a53a0 pc=0xac3f2d
github.com/centrifuge/go-substrate-rpc-client/v4/types.EventRecordsRaw.DecodeEventRecords({0xc0004daa80, 0x4eb, 0xa80}, 0x2393612c904398ee?, {0xc73d20?, 0xc000388800})
        github.com/centrifuge/go-substrate-rpc-client/v4@v4.0.5/types/event_record.go:555 +0x1569 fp=0xc0004a57b0 sp=0xc0004a5448 pc=0xad2529
github.com/threefoldtech/substrate-client-main.(*Substrate).checkForError(0xc0004a5998?, 0xc000416150, 0x686d85?, {0xf, 0xb7, 0x9b, 0x7d, 0x76, 0x65, 0xaa, ...}, ...)
        github.com/threefoldtech/substrate-client-main@v0.0.1/utils.go:236 +0x168 fp=0xc0004a5970 sp=0xc0004a57b0 pc=0xb66328
github.com/threefoldtech/substrate-client-main.(*Substrate).CreateNodeContract(0xc0005ca090, {0x7f358467e480, 0xc000090740}, 0x128000?, {0xc0003407c0, 0x38}, {0xc00003d9c0?, 0xc0004a5bb0?}, 0x40b076?, 0x0)
        github.com/threefoldtech/substrate-client-main@v0.0.1/contract.go:234 +0x3ac fp=0xc0004a5b48 sp=0xc0004a5970 pc=0xb5cc2c
github.com/threefoldtech/terraform-provider-grid/pkg/subi.(*SubstrateMainImpl).CreateNodeContract(0x0?, {0x7f35847c9940?, 0xc000090740}, 0x0?, {0xc0003407c0, 0x38}, {0xc00003d9c0, 0x20}, 0x1?, 0x0)
        github.com/threefoldtech/terraform-provider-grid/pkg/subi/main_substrate.go:52 +0xac fp=0xc0004a5bc0 sp=0xc0004a5b48 pc=0xb9a32c
github.com/threefoldtech/terraform-provider-grid/pkg/deployer.(*DeployerImpl).deploy(0xc000410e40, {0xf8cc68, 0xc000410cc0}, {0xf91130?, 0xc00000e050}, 0x0?, 0x0?, 0x0?)
        github.com/threefoldtech/terraform-provider-grid/pkg/deployer/deployer.go:139 +0x694 fp=0xc0004a69b8 sp=0xc0004a5bc0 pc=0xbd2294
github.com/threefoldtech/terraform-provider-grid/pkg/deployer.(*DeployerImpl).Deploy(0xc000410e40, {0xf8cc68, 0xc000410cc0}, {0xf91130, 0xc00000e050}, 0xc00008e690?, 0xc00008eea0?)
        github.com/threefoldtech/terraform-provider-grid/pkg/deployer/deployer.go:72 +0x108 fp=0xc0004a6a70 sp=0xc0004a69b8 pc=0xbd19c8
github.com/threefoldtech/terraform-provider-grid/internal/provider.(*NetworkDeployer).Deploy(0xc0004a6c78, {0xf8cc68, 0xc000410cc0}, {0xf91130, 0xc00000e050})
        github.com/threefoldtech/terraform-provider-grid/internal/provider/resource_network.go:721 +0xbc fp=0xc0004a6ae0 sp=0xc0004a6a70 pc=0xc0bcdc
github.com/threefoldtech/terraform-provider-grid/internal/provider.resourceNetworkCreate({0xf8cc68, 0xc000410cc0}, 0x0?, {0xc7aaa0?, 0xc0000c8140?})
        github.com/threefoldtech/terraform-provider-grid/internal/provider/resource_network.go:760 +0x26f fp=0xc0004a6e68 sp=0xc0004a6ae0 pc=0xc0c26f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc00024b420, {0xf8cca0, 0xc0003adf20}, 0xd?, {0xc7aaa0, 0xc0000c8140})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:707 +0x12e fp=0xc0004a6ee0 sp=0xc0004a6e68 pc=0xa7c6ce
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00024b420, {0xf8cca0, 0xc0003adf20}, 0xc00028e340, 0xc000162580, {0xc7aaa0, 0xc0000c8140})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0xa7a fp=0xc0004a7260 sp=0xc0004a6ee0 pc=0xa7da1a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000398528, {0xf8cca0?, 0xc0003ade00?}, 0xc0000f1ea0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe3c fp=0xc0004a7500 sp=0xc0004a7260 pc=0xa7635c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000271400, {0xf8cca0?, 0xc0003ad590?}, 0xc0004d88c0)
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:818 +0x574 fp=0xc0004a79d0 sp=0xc0004a7500 pc=0x93a8f4
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xdef5a0?, 0xc000271400}, {0xf8cca0, 0xc0003ad590}, 0xc0004d8850, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170 fp=0xc0004a7a28 sp=0xc0004a79d0 pc=0x921030
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00011e3c0, {0xf902e0, 0xc0003be820}, 0xc000431680, 0xc0003ac150, 0x14fecc0, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1340 +0xd13 fp=0xc0004a7e48 sp=0xc0004a7a28 pc=0x88bf53
google.golang.org/grpc.(*Server).handleStream(0xc00011e3c0, {0xf902e0, 0xc0003be820}, 0xc000431680, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1713 +0xa1b fp=0xc0004a7f68 sp=0xc0004a7e48 pc=0x890f5b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.1/server.go:965 +0x98 fp=0xc0004a7fe0 sp=0xc0004a7f68 pc=0x8899f8
runtime.goexit()
        runtime/asm_amd64.s:1571 +0x1 fp=0xc0004a7fe8 sp=0xc0004a7fe0 pc=0x463f61
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.1/server.go:963 +0x28a

goroutine 1 [select]:
github.com/hashicorp/go-plugin.Serve(0xc000124ea0)
        github.com/hashicorp/go-plugin@v1.4.6/server.go:471 +0x14c8
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.Serve({0xe30325, 0x8}, 0xc0002c2eb0, {0x0, 0x0, 0x0})
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:312 +0xb8a
github.com/hashicorp/terraform-plugin-sdk/v2/plugin.tf5serverServe(0xc000124e40)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/plugin/serve.go:185 +0x518
github.com/hashicorp/terraform-plugin-sdk/v2/plugin.Serve(0xc000124e40)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/plugin/serve.go:125 +0x18d
main.main()
        github.com/threefoldtech/terraform-provider-grid/main.go:60 +0x24e

goroutine 19 [select]:
github.com/hashicorp/go-plugin.(*gRPCBrokerServer).Recv(0xe678b0?)
        github.com/hashicorp/go-plugin@v1.4.6/grpc_broker.go:121 +0x67
github.com/hashicorp/go-plugin.(*GRPCBroker).Run(0xc000128f50)
        github.com/hashicorp/go-plugin@v1.4.6/grpc_broker.go:411 +0x44
created by github.com/hashicorp/go-plugin.(*GRPCServer).Init
        github.com/hashicorp/go-plugin@v1.4.6/grpc_server.go:85 +0x465

goroutine 20 [IO wait]:
internal/poll.runtime_pollWait(0x7f3584850e80, 0x72)
        runtime/netpoll.go:302 +0x89
internal/poll.(*pollDesc).wait(0xc000125380?, 0xc00008a000?, 0x1)
        internal/poll/fd_poll_runtime.go:83 +0x32
internal/poll.(*pollDesc).waitRead(...)
        internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).Read(0xc000125380, {0xc00008a000, 0x1000, 0x1000})
        internal/poll/fd_unix.go:167 +0x25a

Error: The terraform-provider-grid_v1.6.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by MarioBassem** [comment]
@Parkers145  Interesting...

I have a couple of questions:
- I couldn't understand what you meant by saying that you could replicate it in a deployment, could you clarify?
- Does this happen with multiple versions of the plugin, or only with v1.6.3? 

There is a windows version of the plugin in the release that you could also use.

**Comment by Parkers145** [comment]
> @Parkers145  Interesting...
> 
> 
> 
> I have a couple of questions:
> 
> - I couldn't understand what you meant by saying that you could replicate it in a deployment, could you clarify?
> 
> - Does this happen with multiple versions of the plugin, or only with v1.6.3? 
> 
> 
> 
> There is a windows version of the plugin in the release that you could also use.



1.) I deployed an ubuntu vm, installed terraform, and tried to deploy a main.tf

The same issue occurred 

2.) I'm not sure how to try other versions, how do I roll back?

**Comment by coesensbert** [comment]
Encountering the same issue on mainnet for node 5168, although a slightly different error.

main.tf

```
terraform {
  required_providers {
    grid = {
      source = "threefoldtech/grid"
    }
  }
}

provider "grid" {
}

resource "grid_network" "net1" {
    nodes = [5168]
    ip_range = "10.121.0.0/16"
    name = "testingetw1"
    description = "node network"
    add_wg_access = true
}
resource "grid_deployment" "d1" {
  node = 5168
  network_name = grid_network.net1.name
  disks {
    name = "root"
    size = 25
  }
    vms {
    name = "tst2img"
    description ="node img tst"
    flist = "https://hub.grid.tf/tf-official-vms/ubuntu-22.04-lts.flist"
    cpu = 2
    publicip = false
    publicip6 = false
    memory = 2048
    mounts {
        disk_name = "root"
        mount_point = "/data"
    }
    planetary = true
    env_vars = {
      SSH_KEY ="ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDYNeJXJV2FNEwuQz6e0jkKeqRbKWwftBKq+sjSTqa2x"
    }
  }
}
```

Repeatable output:

```
➜  testvm2 terraform apply -parallelism=1 -auto-approve

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # grid_deployment.d1 will be created
  + resource "grid_deployment" "d1" {
      + id                = (known after apply)
      + ip_range          = (known after apply)
      + name              = "vm"
      + network_name      = "testingetw1"
      + node              = 5168
      + solution_provider = 0
      + solution_type     = "Virtual Machine"

      + disks {
          + name = "root"
          + size = 25
        }

      + vms {
          + computedip  = (known after apply)
          + computedip6 = (known after apply)
          + corex       = false
          + cpu         = 2
          + description = "node img tst"
          + env_vars    = {
              + "SSH_KEY" = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDYNeJXJV2FNEwuQz6e0jkKeqRbKWwftBKq+sjSTqa2x"
            }
          + flist       = "https://hub.grid.tf/tf-official-vms/ubuntu-22.04-lts.flist"
          + ip          = (known after apply)
          + memory      = 2048
          + name        = "tst2img"
          + planetary   = true
          + publicip    = false
          + publicip6   = false
          + ygg_ip      = (known after apply)

          + mounts {
              + disk_name   = "root"
              + mount_point = "/data"
            }
        }
    }

  # grid_network.net1 will be created
  + resource "grid_network" "net1" {
      + access_wg_config   = (known after apply)
      + add_wg_access      = true
      + description        = "node network"
      + external_ip        = (known after apply)
      + external_sk        = (known after apply)
      + id                 = (known after apply)
      + ip_range           = "10.121.0.0/16"
      + name               = "testingetw1"
      + node_deployment_id = (known after apply)
      + nodes              = [
          + 5168,
        ]
      + nodes_ip_range     = (known after apply)
      + public_node_id     = (known after apply)
      + solution_type      = "Network"
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + node1_vm1_ip = (known after apply)
  + public_ip    = (known after apply)
  + public_ip6   = (known after apply)
  + wg_config    = (known after apply)
  + ygg_ip       = (known after apply)
grid_network.net1: Creating...
grid_network.net1: Still creating... [10s elapsed]
╷
│ Error: Plugin did not respond
│ 
│   with grid_network.net1,
│   on main.tf line 12, in resource "grid_network" "net1":
│   12: resource "grid_network" "net1" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-grid_v1.6.3 plugin:

fatal error: runtime: out of memory

runtime stack:
runtime.throw({0xe3c1a2?, 0xff8800000?})
        runtime/panic.go:992 +0x71
runtime.sysMap(0xc000c00000, 0xc000435e90?, 0xc000435ef8?)
        runtime/mem_linux.go:189 +0x11b
runtime.(*mheap).grow(0x1529640, 0x7fc3e7?)
        runtime/mheap.go:1413 +0x225
runtime.(*mheap).allocSpan(0x1529640, 0x7fc3e7, 0x0, 0x1)
        runtime/mheap.go:1178 +0x171
runtime.(*mheap).alloc.func1()
        runtime/mheap.go:920 +0x65
runtime.systemstack()
        runtime/asm_amd64.s:469 +0x49

goroutine 37 [running]:
runtime.systemstack_switch()
        runtime/asm_amd64.s:436 fp=0xc00070d008 sp=0xc00070d000 pc=0x461e80
runtime.(*mheap).alloc(0xff87ce000?, 0x7fc3e7?, 0xc5?)
        runtime/mheap.go:914 +0x65 fp=0xc00070d050 sp=0xc00070d008 pc=0x4268c5
runtime.(*mcache).allocLarge(0x44acb2?, 0xff87cc100, 0x1)
        runtime/mcache.go:213 +0x85 fp=0xc00070d0a0 sp=0xc00070d050 pc=0x416da5
runtime.mallocgc(0xff87cc100, 0xd1e0a0, 0x1)
        runtime/malloc.go:1096 +0x5a5 fp=0xc00070d118 sp=0xc00070d0a0 pc=0x40d1a5
runtime.newarray(0xd78360?, 0xc0000fe901?)
        runtime/malloc.go:1291 +0x52 fp=0xc00070d140 sp=0xc00070d118 pc=0x40d6b2
reflect.unsafe_NewArray(0xc0004a6340?, 0xc0000fd584?)
        runtime/malloc.go:1296 +0x19 fp=0xc00070d160 sp=0xc00070d140 pc=0x45db19
reflect.MakeSlice({0xf92b68, 0xc8f880}, 0x7fc3e608, 0x7fc3e608)
        reflect/value.go:2911 +0xac fp=0xc00070d190 sp=0xc00070d160 pc=0x497e2c
github.com/centrifuge/go-substrate-rpc-client/v4/scale.Decoder.DecodeIntoReflectValue({{0xf85880?, 0xc000622a50?}}, {0xc8f880?, 0xc0004a88a8?, 0xc00070d438?})
        github.com/centrifuge/go-substrate-rpc-client/v4@v4.0.5/scale/codec.go:436 +0x48a fp=0xc00070d3a0 sp=0xc00070d190 pc=0xac448a
github.com/centrifuge/go-substrate-rpc-client/v4/scale.Decoder.Decode({{0xf85880?, 0xc000622a50?}}, {0xc0004a6340?, 0xc0004a88a8})
        github.com/centrifuge/go-substrate-rpc-client/v4@v4.0.5/scale/codec.go:325 +0x28d fp=0xc00070d448 sp=0xc00070d3a0 pc=0xac3f2d
github.com/centrifuge/go-substrate-rpc-client/v4/types.EventRecordsRaw.DecodeEventRecords({0xc00093ca80, 0x502, 0xa80}, 0xb0e3a4afb1ea49a8?, {0xc73d20?, 0xc000608800})
        github.com/centrifuge/go-substrate-rpc-client/v4@v4.0.5/types/event_record.go:555 +0x1569 fp=0xc00070d7b0 sp=0xc00070d448 pc=0xad2529
github.com/threefoldtech/substrate-client-main.(*Substrate).checkForError(0xc00070d998?, 0xc00039c0a8, 0x686d85?, {0x55, 0xa9, 0x97, 0x83, 0xe3, 0x53, 0x82, ...}, ...)
        github.com/threefoldtech/substrate-client-main@v0.0.1/utils.go:236 +0x168 fp=0xc00070d970 sp=0xc00070d7b0 pc=0xb66328
github.com/threefoldtech/substrate-client-main.(*Substrate).CreateNodeContract(0xc00039c090, {0x7f8beba9ab98, 0xc0004a6680}, 0x98050?, {0xc000632640, 0x3f}, {0xc0002a4520?, 0xc00070dbb0?}, 0x40b076?, 0x0)
        github.com/threefoldtech/substrate-client-main@v0.0.1/contract.go:234 +0x3ac fp=0xc00070db48 sp=0xc00070d970 pc=0xb5cc2c
github.com/threefoldtech/terraform-provider-grid/pkg/subi.(*SubstrateMainImpl).CreateNodeContract(0x0?, {0x7f8beb9398a0?, 0xc0004a6680}, 0x0?, {0xc000632640, 0x3f}, {0xc0002a4520, 0x20}, 0x1?, 0x0)
        github.com/threefoldtech/terraform-provider-grid/pkg/subi/main_substrate.go:52 +0xac fp=0xc00070dbc0 sp=0xc00070db48 pc=0xb9a32c
github.com/threefoldtech/terraform-provider-grid/pkg/deployer.(*DeployerImpl).deploy(0xc0001589c0, {0xf8cc68, 0xc0001588a0}, {0xf91130?, 0xc00011e038}, 0x0?, 0x0?, 0x0?)
        github.com/threefoldtech/terraform-provider-grid/pkg/deployer/deployer.go:139 +0x694 fp=0xc00070e9b8 sp=0xc00070dbc0 pc=0xbd2294
github.com/threefoldtech/terraform-provider-grid/pkg/deployer.(*DeployerImpl).Deploy(0xc0001589c0, {0xf8cc68, 0xc0001588a0}, {0xf91130, 0xc00011e038}, 0xc0005d3290?, 0xc0005d36e0?)
        github.com/threefoldtech/terraform-provider-grid/pkg/deployer/deployer.go:72 +0x108 fp=0xc00070ea70 sp=0xc00070e9b8 pc=0xbd19c8
github.com/threefoldtech/terraform-provider-grid/internal/provider.(*NetworkDeployer).Deploy(0xc00070ec78, {0xf8cc68, 0xc0001588a0}, {0xf91130, 0xc00011e038})
        github.com/threefoldtech/terraform-provider-grid/internal/provider/resource_network.go:721 +0xbc fp=0xc00070eae0 sp=0xc00070ea70 pc=0xc0bcdc
github.com/threefoldtech/terraform-provider-grid/internal/provider.resourceNetworkCreate({0xf8cc68, 0xc0001588a0}, 0x0?, {0xc7aaa0?, 0xc0000b0640?})
        github.com/threefoldtech/terraform-provider-grid/internal/provider/resource_network.go:760 +0x26f fp=0xc00070ee68 sp=0xc00070eae0 pc=0xc0c26f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000225420, {0xf8cca0, 0xc0005b2e40}, 0xd?, {0xc7aaa0, 0xc0000b0640})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:707 +0x12e fp=0xc00070eee0 sp=0xc00070ee68 pc=0xa7c6ce
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000225420, {0xf8cca0, 0xc0005b2e40}, 0xc0004095f0, 0xc0000d4e80, {0xc7aaa0, 0xc0000b0640})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0xa7a fp=0xc00070f260 sp=0xc00070eee0 pc=0xa7da1a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000400570, {0xf8cca0?, 0xc0005b2d20?}, 0xc000098870)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe3c fp=0xc00070f500 sp=0xc00070f260 pc=0xa7635c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000335360, {0xf8cca0?, 0xc0005b2510?}, 0xc0005d02a0)
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:818 +0x574 fp=0xc00070f9d0 sp=0xc00070f500 pc=0x93a8f4
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xdef5a0?, 0xc000335360}, {0xf8cca0, 0xc0005b2510}, 0xc0005d0230, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170 fp=0xc00070fa28 sp=0xc00070f9d0 pc=0x921030
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000003c0, {0xf902e0, 0xc00040f520}, 0xc000402000, 0xc0003ffc20, 0x14fecc0, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1340 +0xd13 fp=0xc00070fe48 sp=0xc00070fa28 pc=0x88bf53
google.golang.org/grpc.(*Server).handleStream(0xc0000003c0, {0xf902e0, 0xc00040f520}, 0xc000402000, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1713 +0xa1b fp=0xc00070ff68 sp=0xc00070fe48 pc=0x890f5b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.1/server.go:965 +0x98 fp=0xc00070ffe0 sp=0xc00070ff68 pc=0x8899f8
runtime.goexit()
        runtime/asm_amd64.s:1571 +0x1 fp=0xc00070ffe8 sp=0xc00070ffe0 pc=0x463f61
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.1/server.go:963 +0x28a

goroutine 1 [select]:
github.com/hashicorp/go-plugin.Serve(0xc00007ec60)
        github.com/hashicorp/go-plugin@v1.4.6/server.go:471 +0x14c8
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.Serve({0xe30325, 0x8}, 0xc00032cdf0, {0x0, 0x0, 0x0})
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:312 +0xb8a
github.com/hashicorp/terraform-plugin-sdk/v2/plugin.tf5serverServe(0xc00007ec00)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/plugin/serve.go:185 +0x518
github.com/hashicorp/terraform-plugin-sdk/v2/plugin.Serve(0xc00007ec00)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/plugin/serve.go:125 +0x18d
main.main()
        github.com/threefoldtech/terraform-provider-grid/main.go:60 +0x24e

goroutine 7 [select]:
github.com/hashicorp/go-plugin.(*gRPCBrokerServer).Recv(0xe678b0?)
        github.com/hashicorp/go-plugin@v1.4.6/grpc_broker.go:121 +0x67
github.com/hashicorp/go-plugin.(*GRPCBroker).Run(0xc000098fa0)
        github.com/hashicorp/go-plugin@v1.4.6/grpc_broker.go:411 +0x44
created by github.com/hashicorp/go-plugin.(*GRPCServer).Init
        github.com/hashicorp/go-plugin@v1.4.6/grpc_server.go:85 +0x465

goroutine 8 [IO wait]:
internal/poll.runtime_pollWait(0x7f8bebaccef0, 0x72)
        runtime/netpoll.go:302 +0x89
internal/poll.(*pollDesc).wait(0xc00007f140?, 0xc000128000?, 0x1)
        internal/poll/fd_poll_runtime.go:83 +0x32
internal/poll.(*pollDesc).waitRead(...)
        internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).Read(0xc00007f140, {0xc000128000, 0x1000, 0x1000})
        internal/poll/fd_unix.go:167 +0x25a

Error: The terraform-provider-grid_v1.6.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by coesensbert** [comment]
Using provider 1.6.2 works for me.

Used this dirty workaround and copy'd these from a workload that contains 1.6.2 grid provider:
- .terraform.lock.hcl
- .terraform/providers/registry.terraform.io/threefoldtech/grid/1.6.2

**Comment by xmonader** [comment]
1.6.4 been released based on 1.6.2 and will check the out of memory issues on development mode later this week, thanks for your patience. please reopen if your issue is still the same with latest.
